### PR TITLE
Add unit test for models

### PR DESCRIPTION
**Pull request by mohamed20163858**
in this sub-project I added unit tests for all models I created `User`, `Post`, `Comment`, and `Like`  using the RSpec-rails library also I added some validations like the name is not blank, post counter is greater or equal to zero and it must be an integer, Title must not be blank. Title must not exceed 250 characters, CommentsCounter must be an integer greater than or equal to zero, and LikesCounter must be an integer greater than or equal to zero. 
here is a screenshot after passing all tests
image

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
-  has_many :comments, foreign_key: :AuthorId
-  has_many :posts, foreign_key: :AuthorId
-  has_many :likes, foreign_key: :AuthorId
+  has_many :comments, foreign_key: :author_id
+  has_many :posts, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
 
+  validates :name, presence: true, length: { minimum: 3, maximum: 20 }
+  validates :bio, presence: true, length: { minimum: 3, maximum: 150 }
+  validates :posts_counter, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```
Comment (tresorsawasawa) [inline app/models/user.rb]: - [ ] Kindly make sure the validation for the `posts_counter` column is of type `integer` and not `numericality` as required for the project. If you check project the requirements you will find that it is mentioned as follow: 
**PostsCounter must be an integer greater than or equal to zero.**

File: app/models/post.rb [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
-  has_many :likes, foreign_key: :PostId
-  belongs_to :author, class_name: 'User', foreign_key: :AuthorId
+  has_many :comments, foreign_key: :post_id
+  has_many :likes, foreign_key: :post_id
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
   after_save :update_posts_counter
 
+  validates :title, presence: true, length: { minimum: 3, maximum: 250 }
+  validates :comments_counter, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```
Comment (tresorsawasawa) [inline app/models/post.rb]: - [ ] Kindly make sure both `likes_counter` and `comments_counter` are of type `integer` and not `numericality` as required for the project.

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @mohamed20163858 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (mohamed20163858) [reply app/models/user.rb]: Kindly refer to this guide you will find the syntax i wrote is correct [link section 2.9](https://guides.rubyonrails.org/active_record_validations.html#validations-overview)

File: app/models/post.rb (hunk 2 above)
Comment (mohamed20163858) [reply app/models/post.rb]: Kindly refer to this guide you will find the syntax i wrote is correct [link section 2.9](https://guides.rubyonrails.org/active_record_validations.html#validations-overview)

**Review by Davidosky007 — APPROVED** [review]
Hi @mohamed20163858 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Good looking readme file 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (tresorsawasawa) [reply app/models/user.rb]: Hi @mohamed20163858 

You are true

File: app/models/post.rb (hunk 2 above)
Comment (tresorsawasawa) [reply app/models/post.rb]: Hi @mohamed20163858 

You are true

**Pull request merged**